independent-projects/arc/runtime/src/main/java/io/quarkus/arc/BindingsSource.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.arc;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Defines a source of interception binding annotations. When this annotation
+ * is present on an {@link InterceptionProxy} parameter of a CDI producer method,
+ * the interception binding annotations present on the target class are ignored
+ * and instead are read from the class defined by this annotation.
+ * <p>
+ * Interceptor bindings for method named {@code M} with return type {@code R}
+ * and parameter types {@code P1, ..., Pn} on the target class are obtained from
+ * a method with an identical name, return type and parameter types declared
+ * on the bindings source class.
+ * <p>
+ * The annotations that can be meaningfully present on the {@code value} class are:
```

Comment:
_"The annotations that can be meaningfully present..."_ This sentence sounds a bit stilted.

Reply:
Let me rewrite :-)

Reply:
Done.